Foundation/FileManager.swift

```diff
@@ -1393,16 +1417,20 @@ extension FileManager {
                 _current = fts_read(stream)
                 while let current = _current {
                     let filename = NSString(bytes: current.pointee.fts_path, length: Int(strlen(current.pointee.fts_path)), encoding: String.Encoding.utf8.rawValue)!._swiftObject
-                    let skipHidden = (_options.contains(.skipsHiddenFiles) && filename[filename._startOfLastPathComponent] == ".")
 
                     switch Int32(current.pointee.fts_info) {
                         case FTS_D:
-                            if skipHidden || _options.contains(.skipsSubdirectoryDescendants) || (_options.contains(.skipsPackageDescendants) && filename.hasSuffix(".app")) {
+                            let (showFile, skipDescendants) = match(filename: filename, to: _options, isDir: true)
```

Comment:
The tuple return is interesting… I'll allow it. :-)

The main concern is that if the `skipDescendants` check ended up touching the file system or doing some other expensive work, it would be wasted work if you just assign it to `_`.